src/compiler/checker.ts

```diff
@@ -22662,7 +22675,7 @@ namespace ts {
                 const properties = getPropertiesOfObjectType(target);
                 for (const targetProp of properties) {
                     const sourceProp = getPropertyOfType(source, targetProp.escapedName);
-                    if (sourceProp) {
+                    if (sourceProp && !some(sourceProp.declarations, hasSkipDirectInferenceFlag)) {

```

Comment:
Maybe it would be nice to skip this at least in non-services calls, but it would be a lot of plumbing to get `checkMode` through. I can make a local in `createTypeChecker` if it seems worthwhile.

Reply:
I'm pretty sure you could translate the `CheckMode` to an `InferenceFlag` if you'd like.

Reply:
`inferTypes` doesn’t have access to the inference flags, so this doesn’t actually improve the plumbing situation 😕

Reply:
A dummy `InferencePriority` that's unset by the start of inference then? 😅